src/components/application_manager/src/request_info.cc

```diff
@@ -257,6 +257,7 @@ uint32_t RequestInfoSet::RemoveMobileRequests() {
 }
 
 const size_t RequestInfoSet::Size() {
+  sync_primitives::AutoLock lock(this_lock_);
```

Comment:
@dev-gh do you mean in the whole class? I left it so because of the comments in request_info.h in line 257
`  // the lock called this_lock_, since the class represent collection by itself.
  sync_primitives::Lock this_lock_;`